editor/script_editor_debugger.cpp

```diff
@@ -712,27 +712,65 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 		bool warning = err[9];
 		bool e;
 		String time = String("%d:%02d:%02d:%04d").sprintf(vals, &e);
-		String txt = time + " - " + (err[8].is_zero() ? String(err[7]) : String(err[8]));
+		String txt = err[8].is_zero() ? String(err[7]) : String(err[8]);
 
-		String tooltip = TTR("Type:") + String(warning ? TTR("Warning") : TTR("Error"));
-		tooltip += "\n" + TTR("Description:") + " " + String(err[8]);
-		tooltip += "\n" + TTR("Time:") + " " + time;
-		tooltip += "\nC " + TTR("Error:") + " " + String(err[7]);
-		tooltip += "\nC " + TTR("Source:") + " " + String(err[5]) + ":" + String(err[6]);
-		tooltip += "\nC " + TTR("Function:") + " " + String(err[4]);
+		TreeItem *r = error_tree->get_root();
+		if (!r) {
+			r = error_tree->create_item();
+		}
 
-		error_list->add_item(txt, EditorNode::get_singleton()->get_gui_base()->get_icon(warning ? "Warning" : "Error", "EditorIcons"));
-		error_list->set_item_tooltip(error_list->get_item_count() - 1, tooltip);
+		TreeItem *error = error_tree->create_item(r);
+		error->set_collapsed(true);
 
-		int scc = p_data[1];
+		error->set_icon(0, get_icon(warning ? "Warning" : "Error", "EditorIcons"));
+		error->set_text(0, time);
+		error->set_text_align(0, TreeItem::ALIGN_LEFT);
+
+		error->set_text(1, txt);
+
+		String source(err[5]);
+		bool source_is_project_file = source.begins_with("res://");
+		if (source_is_project_file)
+			source = source.get_file();
+
+		txt = source + ":" + String(err[6]);
+		String method = err[4];
+		if (method.length() > 0)
+			txt += " @ " + method + "()";
 
-		Array stack;
-		stack.resize(scc);
-		for (int i = 0; i < scc; i++) {
-			stack[i] = p_data[2 + i];
+		TreeItem *c_info = error_tree->create_item(error);
+		c_info->set_text(0, "<" + TTR(source_is_project_file ? "Source" : "C Source") + ">");
+		c_info->set_text(1, txt);
+		c_info->set_text_align(0, TreeItem::ALIGN_LEFT);
+
+		if (source_is_project_file) {
+			Array meta;
+			meta.push_back(source);
+			meta.push_back(err[6]);
+			error->set_metadata(0, meta);
+			c_info->set_metadata(0, meta);
 		}
 
-		error_list->set_item_metadata(error_list->get_item_count() - 1, stack);
+		int scc = p_data[1];
+
+		for (int i = 0; i < scc; i+=3) {
```

Comment:
https://travis-ci.org/godotengine/godot/jobs/426222693#L533

Reply:
Should be good now.